### ISSUE TITLE
kubectl run should have uid and gid flags

### ISSUE THREAD

**Issue by HariSekhon** [kubernetes/kubernetes#75603]
Re-raise of #33065 as we can't re-open it.

I'm running in to this issue at work and it looks like `kubectl run` is letting me down
```
Error from server (runAsUser not specified): admission webhook "validating-webhook.openpolicyagent.org" denied the request: runAsUser not specified
```
Looks like nobody at work has ever been able to `kubectl run`

**Comment by jakexks** [comment]
@kubernetes/sig-cli-feature-requests

**Comment by mattjmcnaughton** [comment]
@xingzhou @thockin - thoughts on trying to revive https://github.com/kubernetes/kubernetes/pull/35678.

If I'm understanding correctly, it sounds like users who have setup policies requiring all containers to have `RunAsUser` are unable to use `kubectl run` without this feature. To me, thats' a compelling reason to add it.

**Comment by HariSekhon** [comment]
Correct, I am unable to `kubectl run` on our kubernetes cluster which has a security policy of requiring `RunAsUser`.

**Comment by mattjmcnaughton** [comment]
Mhmm yes, I think this all makes sense :) I'll still defer to @thockin re if this is an acceptable reason to add this `uid/gid` flag to `kubectl run`, but my personal vote is that it makes sense to support just the use case you're describing.
